### PR TITLE
More of main.c (with vblank & main game loop)

### PR DESCRIPTION
**Pull request by Cleverking2003**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -2,6 +2,365 @@
...
+        if (!(gUnk_03003670 & 0x1000000)) {
+            if (!(gUnk_03003670 & 0x4000)) {
+                m4aSoundMain();
+            }
+        }
+    }
+
+    if (!(gUnk_03003670 & 0x8000)) {
+        keys = ~REG_KEYINPUT & 0xf;
+        if (keys == 0xf) {
```
Comment (jiangzhengwenjz) [inline src/main.c:284]: use the constants in gba/io_reg.h
```C
#define A_BUTTON        0x0001
#define B_BUTTON        0x0002
#define SELECT_BUTTON   0x0004
#define START_BUTTON    0x0008
```

File: src/main.c [hunk 2]
```diff
@@ -2,6 +2,365 @@
...
+        }
+    }
+    return 1;
+}
+
+void sub_08152694(void) {
+    s8 i;
+    u8 *r7 = gUnk_03002EA0, *sb = gUnk_030035E0, *r8 = gUnk_030036A0;
+    gUnk_030039A8 = gUnk_03002E90;
+    gUnk_03002E90 = (~REG_KEYINPUT & 0x3ff);
```
Comment (jiangzhengwenjz) [inline src/main.c:337]: `KEYS_MASK`

File: include/main.h [hunk 3]
```diff
@@ -1,24 +1,90 @@
-#ifndef GUARD_MAIN_H
-#define GUARD_MAIN_H
-
```
[the remark this answers is not in the document]
Comment (Cleverking2003) [reply include/main.h]: oops, fixed

**Comment by Cleverking2003** [comment]
Should be fixed

**Pull request merged**